docs/docs/plugin.md

```diff
@@ -0,0 +1,166 @@
+- __Plugin Descriptor:__ The descriptor defines which set of plugins should be enabled in one instance of Marathon.
+  
+
+
+## Plugin Interface
+
+A separate Marathon plugin interface jar is published in the following Maven repository with every Marathon release:
+
+```
+http://downloads.mesosphere.io/maven
+```
+
+with following dependency:
+
+SBT:
+
+```
+"mesosphere.marathon" %% "plugin-interface" % "x.x.x" % "provided"
+
+```
+
+Maven:
+
+```
+<dependency>
+    <groupId>mesosphere.marathon</groupId>
+    <artifactId>plugin-interface_2.11</artifactId>
+    <version>x.x.x</version>
+</dependency>
+```
+
+Be sure to always use the same version of the plugin interface as the version of Marathon in production.
+
+
+
+## Extension Functionality
+
+To create an extension, you must implement an extension interface from the plugin interface method and put it in a location available to Marathon so that it can be loaded by the ServiceLoader.
+
+### ServiceLoader
+
+Marathon uses the [ServiceLoader](https://docs.oracle.com/javase/8/docs/api/java/util/ServiceLoader.html) 
+to make a service provider available to a running Marathon instance. For background information on that topic, please see [Creating Extensible Applications](https://docs.oracle.com/javase/tutorial/ext/basics/spi.html).
+
+### Dependencies on other libraries
+
+The writer of the plugin can define dependencies on other libraries and use those dependencies if following requirements are met:
+
+- Marathon already has a dependency to that library. You can see all Marathon dependencies by checking out Marathon and running `sbt dependencyTree`. Such a dependency should be defined as `provided` in your build tool of choice.
+  _Please make sure, you use the exact same version of that library!_
+- The library does not conflict with any library bundled with Marathon (for instance, different byte code manipulation libraries are known to interfere).
+  Such a dependency should be defined as `compile` dependency in your build tool of choice.
+  
+Dependant libraries that are not already provided by Marathon have to be shipped along with the plugin itself.
+ 
+### Packaging
+
+The plugin has to be compiled and packaged as `jar` file.
+Inside that jar file the [ServiceLoader](https://docs.oracle.com/javase/8/docs/api/java/util/ServiceLoader.html)-specific provider-configuration file needs to be included.
+ 
+
+### Logging
+
+Marathon uses `slf4j` for logging. The dependency to this library is already available with the plugin interface package.
+If you use these logger classes, all log statement will be available in Marathon.
+
+### Configuration
+
+Since the plugin mechanism in Marathon relies on the ServiceLoader, the plugin class has to implement a default constructor.
+To pass a certain configuration to the plugin, you can use whatever functionality you like.
```

Comment:
Got it. It's not that I'd like to tell the user how to pass the configuration, only I'd like to use slightly more precise language than "functionality." Would "method" work?